nemo/export/trt_llm/nemo/nemo_ckpt_convert.py

```diff
@@ -0,0 +1,597 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Referrence impl in tensorrt_llm: examples/gpt/nemo_ckpt_convert.py."""
+
+import configparser
+import logging
+import math
+import multiprocessing
+import os
+import shutil
+import typing
+from collections import defaultdict
+from pathlib import Path
+
+import numpy as np
+import tensorstore  # this is important even though not used
+import torch
+import zarr
```

Comment:
Is zarr a dependency of mcore @ericharper ?

Reply:
We have that due to distributed ckpt.